apps/omg/test/support/test_helper.ex

```diff
@@ -158,6 +158,23 @@ defmodule OMG.TestHelper do
     {:ok, full_path, file}
   end
 
+  def call_while(caller, matcher, time \\ 5_000, step \\ 10) do
```

Comment:
this is actually a very unreadable implementation and I think we want to revert to the previous one.

Reply:
to elaborate this: `It's harder to read code than to write it.`
and also `Always code as if the person who ends up maintaining your code is a violent psychopath who knows where you live.`

Reply:
I know these quotes :D
I did not like the previous one more.
what about this
```elixir 
 def call_while_not_true(caller, matcher, time \\ 5_000, step \\ 10)

 def call_while_not_true(caller, matcher, time, step) when time <= 0 do
    :timeout
  end

  def call_while_not_true(caller, matcher, time, step) do
    if matcher.(caller.()) do
      :ok
    else
      Process.sleep(step)
      call_while_not_true(caller, matcher, time - step, step)
    end
  end
```

Reply:
I mean, what is the `reason` for the rewrite? It's working and it's very easy to read and it's three weeks old - have the requirements changed? Your implementation is not even faster (because that's what the PR title is saying it's addressing).

Reply:
`call_until` is better name than 'call_while_not_true',
`pull_client_alarmthis` causing an error that was not visible, I am not sure if it is so clear.

The changes definitely result in faster testing :D after repairing some test. 
not speed up function execution.